Core/include/Acts/TrackFitting/BetheHeitlerApprox.hpp

```diff
@@ -0,0 +1,324 @@
+        }
+
+        weight_sum += ret[i].weight;
+      }
+
+      for (int i = 0; i < NComponents; ++i) {
+        ret[i].weight /= weight_sum;
+      }
+
+      return ret;
+    };
+
+    // Return no change
+    if (x < noChangeLimit) {
+      Array ret(1);
+
+      ret[0].weight = 1.0;
+      ret[0].mean = 1.0;  // p_initial = p_final
+      ret[0].var = 0.0;
+
+      return ret;
+    }
+    // Return single gaussian approximation
+    if (x < singleGaussianLimit) {
+      Array ret(1);
+      ret[0] = BetheHeitlerApproxSingleCmp::mixture(x)[0];
+      return ret;
+    }
+    // Return a component representation for lower x0
+    if (x < lowerLimit) {
+      return make_mixture(m_low_data, x, m_low_transform);
+    }
+    // Return a component representation for higher x0
+    // Cap the x because beyond the parameterization goes wild
+    const auto high_x = std::min(higherLimit, x);
+    return make_mixture(m_high_data, high_x, m_high_transform);
+  }
+
+  /// Loads a parameterization from a file according to the Atlas file
+  /// description
+  ///
+  /// @param low_parameters_path Path to the foo.par file that stores
+  /// the parameterization for low x/x0
+  /// @param high_parameters_path Path to the foo.par file that stores
+  /// the parameterization for high x/x0
+  static auto loadFromFiles(const std::string &low_parameters_path,
+                            const std::string &high_parameters_path) {
+    auto read_file = [](const std::string &filepath) {
+      std::ifstream file(filepath);
+
+      if (!file) {
+        throw std::invalid_argument("Could not open '" + filepath + "'");
+      }
+
+      std::size_t n_cmps, degree;
+      bool transform_code;
+
+      file >> n_cmps >> degree >> transform_code;
+
+      if (NComponents != n_cmps) {
+        throw std::invalid_argument("Wrong number of components in '" +
+                                    filepath + "'");
+      }
+
+      if (PolyDegree != degree) {
+        throw std::invalid_argument("Wrong wrong polynom order in '" +
+                                    filepath + "'");
+      }
+
+      Data data;
+
+      for (auto &cmp : data) {
+        for (auto &coeff : cmp.weightCoeffs) {
+          file >> coeff;
+        }
+        for (auto &coeff : cmp.meanCoeffs) {
+          file >> coeff;
+        }
+        for (auto &coeff : cmp.varCoeffs) {
+          file >> coeff;
+        }
+      }
+
+      return std::make_tuple(data, transform_code);
+    };
+
+    const auto [low_data, low_transform] = read_file(low_parameters_path);
+    const auto [high_data, high_transform] = read_file(high_parameters_path);
+
+    return AtlasBetheHeitlerApprox(low_data, high_data, low_transform,
+                                   high_transform);
+  }
+};
+
+/// Creates a @ref AtlasBetheHeitlerApprox object based on a ATLAS
+/// configuration, that are stored as static data in the source code.
+/// This may not be an optimal configuration, but should allow running
```

Comment:
```suggestion
/// Creates a @ref AtlasBetheHeitlerApprox object based on an ATLAS
/// configuration, that are stored as static data in the source code.
/// This may not be an optimal configuration, but should allow to run
```